### PR TITLE
Basic grpc

### PR DESCRIPTION
**Pull request by man-zhang**
grpc support

### CODE REVIEW

**Comment by man-zhang** [comment]
@arcuri82 the latest commit is only for doc, then request the review.

File: client-java/controller/pom.xml [hunk 1]
```diff
@@ -196,10 +196,54 @@
...
+            <scope>test</scope>
+        </dependency>
+
     </dependencies>
 
     <build>
+
+        <extensions>
+            <extension>
+                <groupId>kr.motd.maven</groupId>
```
Comment (arcuri82) [inline client-java/controller/pom.xml:222]: add a comment specifying what this extension is doing

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/SutController.java [hunk 2]
```diff
@@ -835,8 +835,14 @@ public final void executeAction(RPCActionDto dto, ActionResponseDto responseDto)
                     responseDto.rpcResponse = resSchema.getDto();
                     if (dto.doGenerateAssertions && dto.responseVariable != null)
                         responseDto.assertionScript = resSchema.newAssertionWithJava(dto.responseVariable, dto.maxAssertionForDataInCollection);
-                    else
-                        responseDto.jsonResponse = objectMapper.writeValueAsString(response);
+                    else{
+                        try {
+                            responseDto.jsonResponse = objectMapper.writeValueAsString(response);
+                        }catch (JsonProcessingException e){
+                            // cannot convert to json
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/SutController.java:842]: so if it happens we silently ignore it?

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/schema/types/JavaDtoSpec.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+public enum JavaDtoSpec {
+    /**
+     * pure Java
+     */
+    PURE,
+
+    /**
+     * using proto3
+     */
+    PROTO3
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/schema/types/JavaDtoSpec.java:15]: add bit more info on why it is needed... eg, some examples

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/schema/types/JavaDtoSpec.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package org.evomaster.client.java.controller.problem.rpc.schema.types;
+
+/**
+ * supported specification of Java Dto
+ */
+public enum JavaDtoSpec {
+    /**
+     * pure Java
+     */
+    PURE,
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/schema/types/JavaDtoSpec.java:10]: `PURE` might not be best word here... "pure" functions have a very specific meaning, not sure if confusing then dealing with "pure" dtos.
what about `DEFAULT` ?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

**Comment by man-zhang** [comment]
@arcuri82 there might be a bug in Github. the new commit somehow does not display in this PR (https://github.com/EMResearch/EvoMaster/commits/basic-grpc). If the CI passes, I will close this one, and create another PR in order to include all comments in this branch.

**Pull request closed without merging**